compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -2000,6 +2004,28 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
                   }
                }
 
+            TR::Node *node;
+            ListIterator<TR::Block> blockIt(&blocksInNaturalLoop);
+            TR::Block * derivedInductionVarBlock;
+            for (derivedInductionVarBlock = blockIt.getCurrent(); derivedInductionVarBlock; derivedInductionVarBlock = blockIt.getNext())
+               {
+               for (TR::TreeTopIterator treeTopIt(derivedInductionVarBlock->getEntry(),comp()); treeTopIt != derivedInductionVarBlock->getExit(); treeTopIt.stepForward())
+                    {
+                    node = treeTopIt.currentNode();
+                    if (node->getSymbol() == derivedInductionVar->getSymbol())
+                        goto trivialArrayOperationCheck;
```

Comment:
Do we always expect to hit this path? Is there a possibility we never execute this `goto`?

Reply:
Only time we wouldn't want to execute `trivialArrayOperationCheck` is if we didn't find the node containing the derived induction variable. Don't think that is possible. So based on that, we do always hit that path

Reply:
In that case I'd like to see the assert on line 2021 removed and changed to:

```
TR_ASSERT_FATAL_WITH_NODE(node, node != NULL, "Unable to find node containing the derived induction variable");
```

Reply:
why not just add derivedInductionVarNode to the loop conditions so you don't need that ugly goto?